### PR TITLE
docs: use MDX across documentation

### PR DESCRIPTION
**Pull request by YakovlevCoded**
**Check List**
- [ ] Tests has been run in packages where changes made if available
- [ ] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

[For example #12]

**Description of Changes Made (if issue reference is not provided)**

[Description goes here]

### CODE REVIEW

File: docs/src/components/templates/FrameworkSwitcher.tsx [hunk 1]
```diff
@@ -18,10 +18,10 @@ const FrameworkSwitcher: React.FC<Props> = () => {
 
   function onChange(event: RadioChangeEvent) {
     const framework = event.target.value;
-
+    console.log('env', process.env);
```
Comment (hassankhan) [inline docs/src/components/templates/FrameworkSwitcher.tsx:21]: We can probably remove this line?

Comment (YakovlevCoded) [reply docs/src/components/templates/FrameworkSwitcher.tsx]: Yeah, thank you! I did it in [this](https://github.com/cube-js/cube.js/pull/3362/commits/4d2f6b96b61db8487a7a7e9f8391686bd17f4eb6) commit and fix the framework switcher! Now it's ready for production I guess.

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 2]
```diff
@@ -101,27 +101,11 @@ to `true` on an API instance. This will prevent serving data on API requests fro
...
-configured, Cube.js will only refresh partitions as necessary. Without
-incremental refreshing, Cube.js will re-calculate the entire pre-aggregation
-whenever [the refresh key][ref-schema-ref-preaggs-refresh-key] changes.
-
-<!-- prettier-ignore-start -->
-[[warning |]]
-| Partitioned rollups currently cannot be used by queries without time
-| dimensions.
-<!-- prettier-ignore-end -->
+[Partitioning][wiki-partitioning] is an extremely effective optimization
```
Comment (rpaik) [inline docs/content/Caching/Using-Pre-Aggregations.md]: how about if we say "an effective optimization tool" or "an effective optimization technique"

Comment (rpaik) [reply docs/content/Caching/Using-Pre-Aggregations.md]: Just a minor comment. I'm also OK with leaving the sentence as is 😄

File: docs/netlify.toml [hunk 3]
```diff
@@ -3,38 +3,8 @@ publish = "public/"
...
-
-[[redirects]]
-from = "/*"
-to = "/index.html"
-status = 200
-force = false
+# we need to wrap the public files to the docs folder on prod deployment so we use a different script for this
+[context.production]
+  publish = "dist/"
+  command = "source .env.production && ./build-netlify-prod.sh"
```
Comment (YakovlevCoded) [inline docs/netlify.toml:9]: @hassankhan

**Review by hassankhan — APPROVED** [review]
It's a big one, but it all looks great 💯 

Awesome job, @YakovlevCoded 🙌

**Pull request merged**